client.lua

```diff
@@ -194,32 +207,146 @@ local function StartFueling(vehicle)
+        useWhileDead = false,
+        canCancel = true,
+        disable = {
+            move = true,
+            car = true,
+            combat = true,
+        },
+        anim = {
+            dict = 'timetable@gardener@filling_can',
+            clip = 'gar_ig_5_filling_can',
+            flags = 49,
+        },
+    }, function(cancel)
+        if cancel then
+            isFueling = false
+        end
+    end)
+
+    while isFueling do
+
+        -- TODO: reduce can durability 
+
+        fuel += ox.refillValue
+
+        if(fuel >= 100) then
+            isFueling = false
+            fuel = 100.0
+        end
 
-    if not isNearPump then
-        return notify('Move closer to pump')
+        Wait(ox.refillTick)
     end
 
+    Vehicle:set('fuel', fuel, true)
+    SetVehicleFuelLevel(vehicle, fuel)
+    TriggerServerEvent('ox_fuel:pay', price)
+    -- DEBUG
+    notify(fuel)
+end
+
+local function GetPetrolCan(pumpCoord)
+    local ped = PlayerPedId()
+    local petrolCan = exports.ox_inventory:Search('count', 'WEAPON_PETROLCAN')
+
+    LocalPlayer.state.invBusy = true
+
+    TaskTurnPedToFaceCoord(ped, pumpCoord, ox.petrolCan.duration) 
+    -- Linden broke this changing from entity too coord, needs a better solution
+
+    Wait(500)
+
+    exports.ox_inventory:Progress({
+        duration = ox.petrolCan.duration,
+        label = 'Fueling can',
+        useWhileDead = false,
+        canCancel = true,
+        disable = {
+            move = true,
+            car = true,
+            combat = true,
+            mouse = false
+        },
+        anim = {
+            dict = 'timetable@gardener@filling_can',
+            clip = 'gar_ig_5_filling_can',
+            flags = 49,
+        },
+    }, function(cancel)
+        LocalPlayer.state.invBusy = false
+        if not cancel then
+            if petrolCan > 0 then
+                return TriggerServerEvent('ox_fuel:fuelCan', true, ox.petrolCan.refillPrice)
+            else 
+                return TriggerServerEvent('ox_fuel:fuelCan', false, ox.petrolCan.price)
+            end
+        else
+            return false
+        end
+    end)
+end
+
+RegisterCommand('startfueling', function()
+    local ped = PlayerPedId()
     local vehicle = GetPlayersLastVehicle()
+    local playerCoords = GetEntityCoords(ped)
+    local isNearPump = findClosestPump(playerCoords)
+
+    if not inStation or isFueling or IsPedInAnyVehicle(ped) then return print('skipping fuel -- debug')  end
+    if not isNearPump then return notify('Move closer to pump') end
 
-    if vehicle == 0 or #(GetEntityCoords(vehicle) - playerCoords) > 3 then
+    if not isVehicleCloseEnough(playerCoords, vehicle) and ox.petrolCan.enabled then
+        GetPetrolCan(isNearPump)
+    elseif isVehicleCloseEnough(playerCoords, vehicle) then
+        StartFueling(vehicle)
+    else
         return notify('Vehicle far from pump')
     end
+end)
 
-    StartFueling(vehicle)
+RegisterCommand('startfuelingwithcan', function()
```

Comment:
Again, can probably integrate it into the previous command, why have 2 keybinds that do the same thing